deployability/modules/allocation/aws/provider.py

```diff
@@ -120,23 +131,30 @@ def _load_instance(instance_dir: Path, instance_id: str) -> AWSInstance:
         return AWSInstance(instance_dir, instance_id)
 
     @classmethod
-    def _destroy_instance(cls, instance_dir: str, identifier: str, key_path: str) -> None:
+    def _destroy_instance(cls, instance_dir: str, identifier: str, key_path: str, platform: str, host_identifier: str = None, host_instance_dir: str | Path = None, ssh_port: str = None, arch: str = None) -> None:
         """
         Destroy an AWS EC2 instance.
 
         Args:
             instance_dir (str): Directory where instance data is stored.
             identifier (str): Identifier of the instance.
             key_path (str): Path to the key pair.
+            platform (str): Platform of the instance.
+            host_identifier (str, optional): Identifier of the dedicated host. Defaults to None.
+            host_instance_dir (str | Path, optional): Directory of the host instance. Defaults to None.
+            ssh_port (str, optional): SSH port of the instance. Defaults to None.
+            arch (str, optional): Architecture of the instance. Defaults to None.
```

Comment:
Are all of these parameters required to destroy the instance?

Reply:
It's probably something that I didn't understand correctly, but they have to have the same parameters as the generic/provider, so as I understood all 3 have to have the same parameters (generic/provider, aws/provider and vagrant/provider).

If this is not necessary, it would be good for us to review it so you can explain how to do it correctly.